Fix coach, cable tram and trolleybus GTFS route types

miklcct opened the pull request:
### Summary

Fix GTFS extended route types for coach (2xx) and trolleybus (8xx). Also change the references of "cable car" in the American sense to "cable tram" according to [the reference](https://gtfs.org/documentation/schedule/reference/) as "cable car" has a completely separate meaning (aerial lift) in Europe.

### Issue
Fixes #5550

### Unit tests
added for more existing modes

### Documentation
fixed for cable tram

leonardehrenfried commented:
Due to OTP's complexity, I'm afraid there is no such thing as a trivial change anymore and we expect you to write a test for every PR. Luckily, the testing this is quite easy and you can add more test cases to the existing ones.

On top of this, using COACH instead BUS may be the correct thing to do for the 200-299 type range but it _is_ a breaking change so we will need to check with the relevant users.

leonardehrenfried reviewed (changes requested):
Due to OTP's complexity, I'm afraid there is no such thing as a trivial change anymore and we expect you to write a test for every PR. Luckily, testing this this is quite easy and you can add more test cases to the existing ones.

On top of this, using COACH instead BUS may be the correct thing to do for the 200-299 type range but it is a breaking change so we will need to check with the relevant users.

miklcct commented:
Added change for cable tram as well. GTFS route type 5 was originally called "cable car", but it was a misleading term because a "cable car" means "aerial lift" in Europe, so the specification has now changed to "cable tram".

I have changed all references to it in the UI, and also fixed the Transmodel mapping mistake resulted from the misleading term.

miklcct reviewed (commented):
comment moved to separate lines

t2gran commented:
I am sorry, but I am **not** in favour of merging this PR, it is possible that we can merge piece of it as separate PRs.
 - We have stoped maintaining the old client unless it is a critical fix - we are deleting this code when the new debug client is a good enough alternative for debugging transit (it is already in good shape for debugging street routing).
 - There are two fixes to GTFS mapping and Transmodel in this PR - these changes need to be split into separate PRs. We also need issues for each of these analyzing the consequences and giving people time to fix their data. There are also different strategies for how to do these changes - some without breaking backwards compatibility.

miklcct commented:
Sorry what do you mean? For `CABLE_CAR`, we need to correct the documentation and also the Transmodel mapping because the Transmodel mistake is a direct result of interpreting American English as British English. Should I put these changes into one PR or separate PRs?

For bus / coach, do you think that having a feature flag (I propose to name it as `GtfsCoach`) will be a good idea?

t2gran commented:
> Sorry what do you mean? For CABLE_CAR, we need to correct the documentation and also the Transmodel mapping
> because the Transmodel mistake is a direct result of interpreting American English as British English. Should I put these
> changes into one PR or separate PRs?

No one disagree that it was a mistake, but the mistake is adopted in many clients. So, if someone is selling tickets in one of these clients they might filter on CABLE_CAR - even if it is wrong. So to allow them to migrate we need to analyze if it is a problem first, then find a strategy for migration. This is the "real" task of fixing known problems like this one.

miklcct commented:
I am going to rework on this and release my proposed changes into a few different packages for each mode of transport.

miklcct commented:
Superceded by #6170, #6171, #6173 and #6174 with the exception of the legacy REST API.
